src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -554,7 +554,8 @@ private static ImmutableArray<RefKind> GetEffectiveArgumentRefKinds(ImmutableArr
             // Either the methodOrIndexer is a property, in which case the method used
             // for optional parameters is an accessor of that property (or an overridden
             // property), or the methodOrIndexer is used for optional parameters directly.
-            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor()) ||
+            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && 
+                (optionalParametersMethod.IsAccessor() || ((PropertySymbol)methodOrIndexer).MustCallMethodsDirectly)) ||
```

Comment:
What does `MustCallMethodsDirectly` mean? In what cases it is true?

Reply:
> What does MustCallMethodsDirectly mean? In what cases it is true?

See [https://github.com/dotnet/roslyn/issues/23852](https://github.com/dotnet/roslyn/issues/23852) for more information.

---
In reply to: [157594191](https://github.com/dotnet/roslyn/pull/23840#discussion_r157594191) [](ancestors = 157594191)